examples/BeliefPropagation/ExampleBeliefPropagation.jl

```diff
@@ -0,0 +1,47 @@
+using ITensors
+using ITensorNetworks
+using NamedGraphs
+using DataGraphs
+using MultiDimDictionaries
+using Graphs
+using Random
+using LinearAlgebra
+using UnicodePlots
+
+include("BeliefPropagationFunctions.jl")
+include("../peps/utils.jl")
+
+Random.seed!(1234)
+
+n =14
+z = 3
+chi = 2
+nupdates = 100
+
+println("One Dimensional Chain")
+g = NamedDimGraph(Graphs.SimpleGraphs.grid([n]), [i for i  = 1:n])
+s = siteinds("S=1/2", g)
```

Comment:
I guess we should allow passing a `Graphs.SimpleGraph` to `siteinds` if that isn't supported right now. Also we could make a default constructor `NamedDimGraph(g::SimpleGraph) = NamedDimGraph(g, 1:nv(g))`.